--- a/front-end/src/ui/mod.rs
+++ b/front-end/src/ui/mod.rs
@@ -215,9 +215,9 @@
                 // cpu. And it may be good time to play with unsafe
                 let state_ptr = &mut state_unlocked as *mut std::sync::MutexGuard<'_, State<'_>>;
                 let (mut music_state, mut playlist_state, mut artist_state);
                 unsafe {
-                    music_state = &mut (*state_ptr).musicbar.1;
-                    playlist_state = &mut (*state_ptr).playlistbar.1;
-                    artist_state = &mut (*state_ptr).artistbar.1;
+                    music_state = &mut (&mut *state_ptr).musicbar.1;
+                    playlist_state = &mut (&mut *state_ptr).playlistbar.1;
+                    artist_state = &mut (&mut *state_ptr).artistbar.1;
                 }
 
                 let music_table = MiddleLayout::get_music_container(&mut state_unlocked);
                 screen.render_stateful_widget(music_table, position.music, &mut music_state);
