--- a/src/options.rs	2023-07-16 00:07:22.000000000 +0800
+++ b/src/options.rs	2026-02-02 19:02:08.226803707 +0800
@@ -11,7 +11,7 @@
 //! ```
 
 
-use clap::{Arg, AppSettings};
+use clap::{App, Arg, AppSettings};
 use std::path::PathBuf;
 use std::str::FromStr;
 use term_size;
@@ -59,7 +59,13 @@
             false
         };
 
-        let matches = app_from_crate!("\n")
+        let matches = App::new(env!("CARGO_PKG_NAME"))
+
+            .version(env!("CARGO_PKG_VERSION"))
+
+            .author(env!("CARGO_PKG_AUTHORS"))
+
+            .about(env!("CARGO_PKG_DESCRIPTION"))
             .setting(AppSettings::ColoredHelp)
             .arg(Arg::from_usage("<IMAGE> 'Image file to display'").validator(Options::image_file_validator))
             .arg(szarg)
